src/mgr/DaemonServer.cc

```diff
@@ -249,6 +252,35 @@ bool DaemonServer::ms_dispatch(Message *m)
   };
 }
 
+void DaemonServer::maybe_ready(int32_t osd_id)
```

Comment:
I'm not sure about this one.  Can we see first if we can make the partial PGMap report not result in any errors?  

IMO it is nice to see the fast feedback and watch the % unknown drop to zero.  (Maybe that is more confusing for normal admins, though, than a very laggy update?)